bftengine/tests/bcstatetransfer/bcstatetransfer_tests.cpp

```diff
@@ -630,6 +661,39 @@ TEST_F(BcStTest, dstFullStateTransfer) {
   ASSERT_EQ(BCStateTran::FetchingState::NotFetching, stateTransfer_->getFetchingState());
 }
 
+// TBD BC-14432
+TEST_F(BcStTest, sendPrePrepareMsgsDuringStateTransfer) {
+  bftEngine::ReservedPagesClientBase::setReservedPages(&res_pages_mock_);
+  std::once_flag once_flag;
+  ASSERT_NO_FATAL_FAILURE(SendCheckpointSummaries());
+  mockedSrc_->ReplyAskForCheckpointSummariesMsg();
+  auto ss = GetSourceSelector();
+  while (true) {
+    ASSERT_NO_FATAL_FAILURE(
+        AssertFetchBlocksMsgSent(testParams_.expectedFirstRequiredBlockNum, testParams_.expectedLastRequiredBlockNum));
+    mockedSrc_->ReplyFetchBlocksMsg();
+    if (testParams_.expectedLastRequiredBlockNum <= config_.maxNumberOfChunksInBatch) break;
+    testParams_.expectedLastRequiredBlockNum -= config_.maxNumberOfChunksInBatch;
+    this_thread::sleep_for(chrono::milliseconds(20));
+    onTimerImp();
+
+    std::call_once(once_flag, [&] {
+      // For 7 intervals of itemData msgs we need to send 10 PrePrepare so that source gets changed.
+      auto msg = msg_generator_->GeneratePrePrepareMsg(ss.currentReplica());
+      for (uint16_t i = 1; i <= 10; i++) {
```

Comment:
The number 7 can be ommitted, we might change the range collected. I assume you meant to say "for 7 batches"
Can write: "Generate prePrepare messages to trigger source selector to change the source to avoid primary "
Also:
I think it's not a good practice to put explicit numbers here.
You have MAX_PREPREPARE_MSGS (which might become config param , see other comment).
Extract it and use it - same comment for the rest of the tests written, some are using 9.

Reply:
Thanks. Addressed.